pkg/credentialprovider/identity_bindings_credentials.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentialprovider
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	providerconfig "sigs.k8s.io/cloud-provider-azure/pkg/provider/config"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"k8s.io/klog/v2"
+	v1 "k8s.io/kubelet/pkg/apis/credentialprovider/v1"
+)
+
+const (
+	// Kubernetes certificate paths
+	KubernetesCACertPath = "/etc/kubernetes/certs/ca.crt"
+)
+
+// identityBindingsTokenCredential implements azcore.TokenCredential interface
+// using identity bindings token exchange
+type identityBindingsTokenCredential struct {
+	req           *v1.CredentialProviderRequest
```

Comment:
Could you extract just the required fields instead of including the entire request here?

fyi, we have a PR to add IB mode to the SDK. It would be easier to migrate if you keep the Azure credential a bit more independent from the credential provider. https://github.com/Azure/azure-sdk-for-go/pull/25664

example use: https://github.com/Azure/secrets-store-csi-driver-provider-azure/compare/master...clnv:secrets-store-csi-driver-provider-azure:ib#diff-0ccbf9a2a7d709b87dbf2f30cc730528c6299d537bf721300d49502cfe521c0aR199-R220

Reply:
I noticed Billy's change in this file does not pass `tenantID` to IB proxy, so my assumption is IB proxy does not need tenant ID to work, but the `azidentity` library requires `tenantID`?

Reply:
yes. though the underlying impl might not use it directly, it’s still required.

Reply:
I see. Then let's also add a parameter `--ib-default-tenant-id` and always pass this argument since day 1. This will make our transition much smoother if some day the acr credential provider switches the use official SDK.